easybuild/easyconfigs/m/Mothur/Mothur-1.48.2_fix-hardcoding.patch

```diff
@@ -0,0 +1,118 @@
+ endif
+@@ -109,7 +109,7 @@
+     subdirs :=  $(sort $(dir $(filter-out  $(skipUchime), $(wildcard source/*/))))
+     subDirIncludes = $(patsubst %, -I %, $(subdirs))
+     subDirLinking =  $(patsubst %, -L%, $(subdirs))
+-    CXXFLAGS += -I. $(subDirIncludes)
++    CXXFLAGS += -I. $(subDirIncludes) -Isource
+     LDFLAGS += $(subDirLinking)
+ 
+ 
+@@ -118,6 +118,9 @@
+ #
+     OBJECTS=$(patsubst %.cpp,%.o,$(wildcard $(addsuffix *.cpp,$(subdirs))))
+     OBJECTS+=$(patsubst %.c,%.o,$(wildcard $(addsuffix *.c,$(subdirs))))
++# Include additional objects in source/
++    OBJECTS+=$(patsubst %.cpp,%.o,$(wildcard source/*.cpp))
++    OBJECTS+=$(patsubst %.c,%.o,$(wildcard source/*.c))
+     OBJECTS+=$(patsubst %.cpp,%.o,$(wildcard *.cpp))
+     OBJECTS+=$(patsubst %.c,%.o,$(wildcard *.c))
+ 
+diff -ruN mothur-1.48.2.orig/makefile-internal /tmp/eb/Mothur/1.48.2/foss-2023b/mothur-1.48.2/makefile-internal
+--- mothur-1.48.2.orig/makefile-internal	2024-11-04 20:53:24.000000000 +0100
++++ /tmp/eb/Mothur/1.48.2/foss-2023b/mothur-1.48.2/makefile-internal	2024-11-15 10:28:51.863460631 +0100
+@@ -43,8 +43,8 @@
+ #The boost libraries allow you to read gz files.
+ ifeq  ($(strip $(USEBOOST)),yes)
+     #statically link so the boost install is not required on users machine
+-    BOOST_INCLUDE_DIR="/usr/local/include"
+-    BOOST_LIBRARY_DIR="/usr/local/lib"
++    BOOST_INCLUDE_DIR="${EBROOTBOOST}/include"
++    BOOST_LIBRARY_DIR="${EBROOTBOOST}/lib"
+ 
+     #windows paths
+     #BOOST_INCLUDE_DIR="/usr/x86_64-w64-mingw32/sys-root/mingw/include"
+@@ -62,8 +62,8 @@
+ #User specified HDF5 library
+ ifeq  ($(strip $(USEHDF5)),yes)
+ 
+-    HDF5_INCLUDE_DIR="/usr/local/include"
+-    HDF5_LIBRARY_DIR="/usr/local/lib"
++    HDF5_INCLUDE_DIR="${EBROOTHDF5}/include"
++    HDF5_LIBRARY_DIR="${EBROOTHDF5}/lib"
+ 
+     LDFLAGS +=  -L ${HDF5_LIBRARY_DIR} 
+     LIBS += ${HDF5_LIBRARY_DIR}/libhdf5_hl_cpp.a
+@@ -110,6 +110,9 @@
+ #
+     OBJECTS=$(patsubst %.cpp,%.o,$(wildcard $(addsuffix *.cpp,$(subdirs))))
+     OBJECTS+=$(patsubst %.c,%.o,$(wildcard $(addsuffix *.c,$(subdirs))))
++    # Include additional objects in source/
++    OBJECTS+=$(patsubst %.cpp,%.o,$(wildcard source/*.cpp))
++    OBJECTS+=$(patsubst %.c,%.o,$(wildcard source/*.c))
+     OBJECTS+=$(patsubst %.cpp,%.o,$(wildcard *.cpp))
+     OBJECTS+=$(patsubst %.c,%.o,$(wildcard *.c))
+ 
+Binary files mothur-1.48.2.orig/source/calculators/prng.o and /tmp/eb/Mothur/1.48.2/foss-2023b/mothur-1.48.2/source/calculators/prng.o differ
+diff -ruN mothur-1.48.2.orig/source/uchime_src/makefile /tmp/eb/Mothur/1.48.2/foss-2023b/mothur-1.48.2/source/uchime_src/makefile
+--- mothur-1.48.2.orig/source/uchime_src/makefile	2024-11-04 20:53:24.000000000 +0100
++++ /tmp/eb/Mothur/1.48.2/foss-2023b/mothur-1.48.2/source/uchime_src/makefile	2024-11-15 10:29:38.800547895 +0100
+@@ -1,4 +1,4 @@
+-CXXFLAGS = -O3 -D_FILE_OFFSET_BITS=64 -DNDEBUG=1 -DUCHIMES=1
++CXXFLAGS = -O3 -D_FILE_OFFSET_BITS=64 -DNDEBUG=1 -DUCHIMES=1 -std=c++11
```

Comment:
Hm, i wonder if we can just make this add the -D flags instead of overwriting CXXFLAGS completely, so that it catches the EB environment variables which set compiler flags (though I'm not going to insist of it in any way gets tricky).

Reply:
Very sorry, but I'm not sure how to do this. This would be in the .eb file? thanks